Made how-to-play-subsets.md for solving issue #94

SuperGamer18 opened the pull request:
https://github.com/SuperGamer18/RetroAchievements-docs/blob/updates/docs/general/tutorials/how-to-play-subsets.md
This is the first time I make an entirely new page for the RADocs. Also the image quality may not be the best, sorry. It features a brief explanation about subsets and an easy to understand guide (assuming you already have a general idea about hashes). Yes, I am aware that people may use other options for patching ROMs, such as Lunar IPS, but I think Rom Patcher JS is the easiest and most intuitive one to use.

SuperGamer18 committed:
Added info about some more emulators and polished a bit

SuperGamer18 committed:
Merge branch 'main' into patch-1

SuperGamer18 committed:
Update emulator-support-and-issues.md

Fixed version requested by @wescopeland

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Update emulator-support-and-issues.md

Fixed yet again

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Updated the translation to be on par with the English version and made it more accurate

SuperGamer18 committed:
Merge pull request #1 from SuperGamer18/patch-1

Updated the translation to be on par with the English version [...]

SuperGamer18 committed:
Heavily improved from what was shown in the initial PR

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Hopefully the final fix before the Spanish translation is complete

SuperGamer18 committed:
For real now

SuperGamer18 committed:
Finally finished it to be on par with the current English version

SuperGamer18 committed:
Merge branch 'RetroAchievements:main' into updates

SuperGamer18 committed:
Merge branch 'RetroAchievements:main' into updates

SuperGamer18 committed:
GitHub was down. Bummer.

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Merge branch 'RetroAchievements:main' into updates

SuperGamer18 committed:
Heavy simplification as requested

SuperGamer18 committed:
Fixed the zone name in the Titles section and other minor fixes

For clarification, the name "Green Hills Zone" is only used in the 8-bit versions of Sonic 2, which while looking similar, is actually different from the iconic zone, whose correct name that should be used in the example shown here is "Green Hill Zone", without the s.

SuperGamer18 committed:
Merge pull request #2 from SuperGamer18/patch-1

Fixed the zone name in the Titles section and other minor fixes

SuperGamer18 committed:
Added Frontmatter and made more small fixes

SuperGamer18 committed:
Oxford comma

SuperGamer18 committed:
Update set-plagiarism.md

SuperGamer18 committed:
Update set-plagiarism.md

SuperGamer18 committed:
Reworded Frontmatter, now just 165 characters

SuperGamer18 committed:
Merge branch 'updates' into patch-1

SuperGamer18 committed:
Merge pull request #3 from SuperGamer18/patch-1

Reworded Frontmatter + Oxford Comma

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Yet again, made it on par with the English version.

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Update emulator-support-and-issues.md

SuperGamer18 committed:
Removed unnecessary line breaks and slight polished it

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Temporally reverted, will remake it soon

SuperGamer18 committed:
Fixed ³

SuperGamer18 committed:
Update set-plagiarism.md

SuperGamer18 committed:
Pequeñas correcciones de traducción

SuperGamer18 committed:
Forgot Arduboy

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 committed:
Fixed Intellivision, again

SuperGamer18 committed:
Merge branch 'RetroAchievements:main' into updates

SuperGamer18 committed:
Added how-to-play-subsets.md

SuperGamer18 committed:
Added how-to-play-subsets.md

SuperGamer18 committed:
Added images and other stuff

SuperGamer18 committed:
Update how-to-play-subsets.md

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
A good start.

It looks like something may be funky with the image attachments. Those also need to exist as new files in the repo. We typically put these in the _docs/public_ folder.

For this change, it may be worth playing with the "Detailed Contributions" guide: https://docs.retroachievements.org/general/contributing-to-the-docs.html#detailed-contributions

SuperGamer18 replied:
https://github.com/SuperGamer18/RetroAchievements-docs/tree/updates/docs/public uploaded the files, what should I do now?

wescopeland replied:
Much better :)

SuperGamer18 committed:
Better crop for "Apply patch"

SuperGamer18 committed:
Update how-to-play-subsets.md

SuperGamer18 committed:
Added files for how-to-play-subsets.md

wescopeland commented on docs/public/rom-patcher-js.png:
This should be in English as it'll be shown in the English docs.

SuperGamer18 replied:
Fair point, but the site's language was set to English. Maybe I should try setting the browser's language to English as well...

wescopeland commented on docs/public/retroachievements-hashes.png:
For this screenshot, can we use the default theme? That's what most users will be accustomed to seeing; color customization is more of a power user feature, so most people may be confused by this.

SuperGamer18 replied:
Yeah, there might be users who may not see what's at the bottom of the page.

SuperGamer18 committed:
Replaced rom-patcher-js.png and retroachievements-hashes.png

SuperGamer18 committed:
Let's try that again

SuperGamer18 commented:
Both images have been fixed.

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+description: Brief explanation of subsets with a simple tutorial on how to patch a ROM for a subset.
+---
+
+# What are Subsets?
```
This should match the frontmatter title:
```suggestion
# How to Play Subsets
```

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+# What are Subsets?
+
+Subsets are typically extra sets for a game when certain achievements would just not fit into the main set. There are many types of subsets, but for more info on that topic see [this article](https://docs.retroachievements.org/guidelines/content/subsets.html#types-of-subsets). Subsets typically require an extra patch or sometimes just an alternate revision/version of the game.
```
```suggestion
Subsets are supplemental achievement sets for a game that contain achievements which may not be a good fit into the main (core) set. There are [many types of subsets](/guidelines/content/subsets.html#types-of-subsets). Subsets typically require an extra patch or sometimes an alternate revision/version of the game.
```

A few edits here for brevity. Also changed the link from absolute ("https://docs.retroachievements.org...") to relative ("/guidelines/content/..."). With relative links, the automated check that runs on PRs can detect if the link is broken and throw an error, which saves us a lot of pain and suffering as the docs get bigger and bigger.

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+Subsets are typically extra sets for a game when certain achievements would just not fit into the main set. There are many types of subsets, but for more info on that topic see [this article](https://docs.retroachievements.org/guidelines/content/subsets.html#types-of-subsets). Subsets typically require an extra patch or sometimes just an alternate revision/version of the game.
+
+## How to Play Subsets
```
Since the title will be "How to Play Subsets", this should be changed to something else. Maybe "Getting Started"?

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+**Step 1:** Download the patch for the subset of your choice. We'll use [this subset](https://retroachievements.org/game/7596) as an example (in this case, Super Mario Land [Subset - Bonus]). Patches are hosted [here](https://github.com/RetroAchievements/RAPatches).
+
+![RetroAchievements hashes](https://github.com/user-attachments/assets/7221c0da-54f3-4480-8f2a-3a55c8f74f13)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+**Step 2:** Go to [this site](https://www.marcrobledo.com/RomPatcher.js/).
+
+![Rom Patcher JS](https://github.com/user-attachments/assets/af91a240-fd2a-4a00-82b9-6ef9b675b960)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+**Step 3:** Choose your ROM in the upper field and your patch file in the lower one. Make sure that you are using the correct ROM for the patch (see the readme file that usually comes included with the patch itself). Checking the CRC32, MD5 and SHA-1 hashes between both fields also helps (in this case, the subset offers patches for both the original, unmodified Super Mario Land and the Super Mari ...
+
+![ROM field](https://github.com/user-attachments/assets/bd763c82-bab6-47f6-958c-56aa90113066)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+![ROM field](https://github.com/user-attachments/assets/bd763c82-bab6-47f6-958c-56aa90113066)
+
+![Patch field](https://github.com/user-attachments/assets/b158a568-b6ac-49f7-9473-6807ad638fe7)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+**Step 4:** Click/tap on "Apply patch" and download your patched ROM (to distinguish it from the original ROM, we recommend naming it like the subset or using "Use patch name for output").
+
+![Apply patch button](https://github.com/user-attachments/assets/a19f88a0-9233-4836-9622-2f5adbd05cbc)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+**Step 5:** Open your emulator of choice, then load your patched ROM, and see if a pop-up appears telling you that you're using "X Game [Subset - X]". If so, congratulations! You have properly patched your ROM! Otherwise, repeat all the previous steps.
+
+![RetroArch Bonus set](https://github.com/user-attachments/assets/19e5a861-7240-4ed2-b472-ba0bf3143836)
```
This image needs to link to the file in the public folder (check other pages with images for examples).

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+## How to Play Subsets
+
+In case that the subset you want to play requires a patch, follow these steps below. Similar steps can be followed when patching a ROM to play a hack.
```
```suggestion
When the subset you want to play requires a patch, follow these steps. Similar steps can be followed when patching a ROM to play a hack.
```

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+In case that the subset you want to play requires a patch, follow these steps below. Similar steps can be followed when patching a ROM to play a hack.
+
+**Step 1:** Download the patch for the subset of your choice. We'll use [this subset](https://retroachievements.org/game/7596) as an example (in this case, Super Mario Land [Subset - Bonus]). Patches are hosted [here](https://github.com/RetroAchievements/RAPatches).
```
```suggestion
**Step 1:** Download the patch for the subset of your choice. We'll use [this subset](https://retroachievements.org/game/7596) as an example (in this case, Super Mario Land [Subset - Bonus]). Patches are hosted on [RAPatches](https://github.com/RetroAchievements/RAPatches).
```

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+![RetroAchievements hashes](https://github.com/user-attachments/assets/7221c0da-54f3-4480-8f2a-3a55c8f74f13)
+
+**Step 2:** Go to [this site](https://www.marcrobledo.com/RomPatcher.js/).
```
```suggestion
**Step 2:** Go to [ROM Patcher JS](https://www.marcrobledo.com/RomPatcher.js/), a free online tool to patch your ROM.
```

wescopeland commented on docs/general/tutorials/how-to-play-subsets.md:
```diff
@@ -0,0 +1,36 @@
+
+![Patch field](https://github.com/user-attachments/assets/b158a568-b6ac-49f7-9473-6807ad638fe7)
+
+**Step 4:** Click/tap on "Apply patch" and download your patched ROM (to distinguish it from the original ROM, we recommend naming it like the subset or using "Use patch name for output").
```
```suggestion
**Step 4:** Click/tap on "Apply patch" and download your patched ROM.
```
Safe to omit this sentence.

SuperGamer18 committed:
Update how-to-play-subsets.md

SuperGamer18 committed:
Almost done, will address the image issue soon

SuperGamer18 committed:
Merge branch 'RetroAchievements:main' into updates

SuperGamer18 commented:
I've addressed all issues so far, except for the image ones. I've searched through all the pages in the tutorials folder and haven't found anything to help me. Would you mind linking me to one or two pages that contain your preferred solution? (I'm not sure what other pages contain images and this is the first time I ever have to do this btw)

wescopeland commented:
Any page which one of the existing images should do, eg: _docs/developer-docs/developer-welcome.md_.

SuperGamer18 committed:
Merge branch 'main' into updates

SuperGamer18 commented:
I see, not that complex.

SuperGamer18 committed:
Different method of image linking

SuperGamer18 commented:
That's weird... The first time this ever failed...

wescopeland commented:
It failing is a good safeguard to have - make extra sure the image path in the new doc file is correct.

SuperGamer18 commented:
Just checked, all images load properly. And I guess you're telling me I did everything correctly, right?

wescopeland commented:
Looks like there is a build issue:
> docs/public/retroachievements-hashes.png

I think maybe this should instead be /public/retroachievements-hashes.png

SuperGamer18 commented:
That doesn't work correctly for me.
![Screenshot](https://github.com/user-attachments/assets/1e246989-34ca-4f2f-9e14-c2ebc81d35a5)

wescopeland commented:
We can't accept any PR where checks are failing.

"docs/public/*" is very clearly the incorrect import path. Please see all other image imports, eg: _emulator-setup-for-developers.md_.

SuperGamer18 committed:
Update how-to-play-subsets.md

SuperGamer18 commented:
Done, but keep in mind that I'll no longer be able to preview the images directly from the file or whatever. At least it doesn't fail this time. (Also how is "docs/public/*" the incorrect import path?)

wescopeland commented:
> how is "docs/public/*" the incorrect import path?

Browsers do not understand markdown files, so an intermediate step is needed to convert the markdown files to normal HTML that browsers do understand. This happens on every redeploy of the docs site when a change is made (such as when a PR gets merged).

The intermediate step that does this is powered by something called [VitePress](https://vitepress.dev/).

The check that runs on every commit is VitePress scanning over all the files in the project and giving a thumbs up or a thumbs down if it's happy with what it sees.

Web servers generally put static files like images in their _public_ directory. This is why all the links conventionally need to point to _public/something.png_.

SuperGamer18 commented:
Fair point. Anything else I should do?

wescopeland commented:
Nope, looks good! Thanks. I'll deploy the change.
